inference-engine/src/transformations/src/transformations/common_optimizations/compress_constants.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/compress_constants.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::CompressConstants, "CompressConstants", 0);
+
+ngraph::pass::CompressConstants::CompressConstants() {
+    MATCHER_SCOPE(CompressConstants);
+    auto const_node = ngraph::pattern::wrap_type<opset8::Constant>();
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto& const_node_pattern = pattern_map.at(const_node);
+
+        auto const_node = std::dynamic_pointer_cast<ngraph::opset8::Constant>(
+            const_node_pattern.get_node_shared_ptr());
+
+        if (const_node->get_element_type() != ov::element::f32 && const_node->get_element_type() != ov::element::f64)
+            return false;
+
+        const auto& rt_info = const_node->get_rt_info();
+        if (rt_info.count("DISABLE_FP16_COMPRESSION"))
+            return false;
+
+        auto new_const = std::make_shared<ngraph::opset8::Constant>(ov::element::f16,
+            const_node->get_shape(),
+            const_node->cast_vector<float16>().data());
+        auto convert = std::make_shared<ngraph::opset8::Convert>(new_const, ov::element::f32);
+
+        convert->set_friendly_name(m.get_match_root()->get_friendly_name());
+        ngraph::copy_runtime_info(const_node, convert);
+        ngraph::replace_node(m.get_match_root(), convert);
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(const_node, matcher_name);
+    this->register_matcher(m, callback);
+}
```

Comment:
can we add tests for implemented transformations?

Reply:
Yeah, I wrote, but forgot to add fies